gravitee-am-test/specs/management/preview-themes.jest.spec.ts

```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import fetch from "cross-fetch";
+import * as faker from 'faker';
+import {afterAll, beforeAll, expect} from '@jest/globals';
+import {requestAdminAccessToken} from "@management-commands/token-management-commands";
+import {createDomain, deleteDomain, startDomain} from "@management-commands/domain-management-commands";
+import { preview } from "@management-commands/form-management-commands";
+
+const path = require("path");
+const fs = require("fs");
+
+global.fetch = fetch;
+
+let accessToken;
+let domain;
+
+let customDraftTheme = {
+    name: faker.address.country(),
+    locale: faker.address.countryCode(),
+    primaryTextColorHex: "#FFFFFF",
+    primaryButtonColorHex: "#000000",
+    secondaryTextColorHex: "#FFFFFF",
+    secondaryButtonColorHex: "#000000"
+};
+
+beforeAll(async () => {
+    const adminTokenResponse = await requestAdminAccessToken();
+    accessToken = adminTokenResponse.body.access_token;
+    expect(accessToken).toBeDefined()
+
+    const createdDomain = await createDomain(accessToken, "domain-themes-preview", faker.company.catchPhraseDescriptor());
+    expect(createdDomain).toBeDefined();
+    expect(createdDomain.id).toBeDefined();
+
+    const domainStarted = await startDomain(createdDomain.id, accessToken);
+    expect(domainStarted).toBeDefined();
+    expect(domainStarted.id).toEqual(createdDomain.id);
+
+    domain = domainStarted;
+});
+
+async function testRequestPreview(template: String, content: String, theme?: any) {
+    const previewResult = await preview(domain.id, accessToken, {
+        type: 'FORM',
+        template: template, 
+        content: content,
+        theme: theme
+    })
+    expect(previewResult).toBeDefined();
+    return previewResult;
+}
+
+const TEMPLATE_DIRECTORY = "../../../gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/resources/webroot/views/";
+
+describe("Testing preview form api...", () => {
+    describe("With LOGIN template", () => {
```

Comment:
These `describes` could be replaced with a method call, e.g:

```
const templateTest = (description: string, template: string) => {
        describe(description, () => {
            const file = path.join(__dirname,
                TEMPLATE_DIRECTORY,
                template + "html");
            const content = fs.readFileSync(file, "utf8", function (err: any, data: any) {
                return data;
            });

            requestPreview(template, content);
        });
    };

//This could be called in a for each/of instead with a map or array of test/template values
templateTest("With LOGIN template", "login");
templateTest("With ERROR template", "error");
templateTest("With RESET_PASSWORD template", "reset_password");

```